### PR TITLE
Add Section tests

### PR DESCRIPTION
**Pull request by itssumertime**
This tests that sections have a title and questions.

### CODE REVIEW

File: client/src/components/Section.test.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import React from "react";
+import Section from "./Section.jsx";
+import { renderWith } from "../util/testing";
+
+expect.extend(toHaveNoViolations);
+
+describe("section", () => {
+  beforeEach(() => {
+    const initialState = {
+      form: [
```
Comment (kebrahim) [inline client/src/components/Section.test.js:13]: you don't need to specify `form` or `responses` because they're not used by Section.jsx

File: client/src/components/Section.test.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+    renderWith(
+      <Section sectionKey={sectionKey} questionKeys={questionKeys} />,
+      {
+        state: initialState,
+      }
+    );
+  });
+
+  it("section is accessible", async () => {
```
Comment (kebrahim) [inline client/src/components/Section.test.js:49]: remove the "section" here. it'll inherit the "section" from the `describe` method above

File: client/src/components/Section.test.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    const { container } = renderWith(<Section />);
```
Comment (kebrahim) [inline client/src/components/Section.test.js:50]: you don't need to call renderWith again, it was already called in the beforeEach method. you might just have to save it to a {container} above so you can access it here

File: client/src/components/Section.test.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    expect(await axe(container)).toHaveNoViolations();
+  });
+
+  it("section title and heading", async () => {
```
Comment (kebrahim) [inline client/src/components/Section.test.js:54]: s/section/renders

File: client/src/components/Section.test.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    const section = screen.getByText("HOUSEHOLD_SECTION.title");
```
Comment (kebrahim) [inline client/src/components/Section.test.js:55]: i don't think you need lines 55-56. you're already verifying the heading appears in lines 57-60

File: client/src/components/Section.test.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    expect(section).toBeInTheDocument();
+    const heading = screen.getByRole("heading", {
+      name: "HOUSEHOLD_SECTION.title",
+    });
+    expect(heading).toBeInTheDocument();
+  });
+
+  it("questions", async () => {
```
Comment (kebrahim) [inline client/src/components/Section.test.js:63]: s/questions/renders questions

File: client/src/components/Section.test.js [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    const questionOne = screen.getByText("ADULTS_65_PLUS.text");
+    expect(questionOne).toBeInTheDocument();
+    const questionTwo = screen.getByText("HOUSEHOLD_SIZE.text");
+    expect(questionTwo).toBeInTheDocument();
+    const questionThree = screen.getByText("HOUSEHOLD_INCOME.text");
+    expect(questionThree).toBeInTheDocument();
+  });
+
+  it("radio buttons", async () => {
```
Comment (kebrahim) [inline client/src/components/Section.test.js:72]: how about "renders radio buttons for boolean question"?

File: client/src/components/Section.test.js (hunk 2 above)
Comment (kebrahim) [inline client/src/components/Section.test.js:49]: mind also adding a keyboard-tab-navigation test that confirms you can navigate from question to question? you can see an example here: https://github.com/ny/maslow/blob/master/client/src/components/Form.test.js#L308

File: client/src/components/Section.test.js (hunk 1 above)
Comment (itssumertime) [reply client/src/components/Section.test.js:13]: I actually get "can't read the property of undefined" and also the sections are within "form". Even removing responses gives the same error, although I'm not using it, so I think it may be something with "renderWith" and Redux. It errors on the useSelector part of the Section.jsx file.

Comment (mscharfstein) [reply client/src/components/Section.test.js:13]: yeah you need a least an empty responses (responses: {}) because NumberQuestion and BooleanQuestion will be called and those will try to look up a key in responses. you shouldnt need form i don't think

File: client/src/components/Section.test.js (hunk 5 above)
Comment (mscharfstein) [reply client/src/components/Section.test.js:55]: +1

File: client/src/components/Section.test.js [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+    const radioYes = screen.getByRole("radio", { name: "radio.yes" });
+    expect(radioYes).toBeInTheDocument();
+    const radioNo = screen.getByRole("radio", { name: "radio.no" });
+    expect(radioNo).toBeInTheDocument();
+  });
+});
```
Comment (mscharfstein) [inline client/src/components/Section.test.js:78]: can you add tests where the showErrorForNoEntry prop is true and theres just an empty response object so the answers are not valid: validate that the error text shows up

File: client/src/components/Section.test.js (hunk 3 above)
Comment (itssumertime) [reply client/src/components/Section.test.js]: Moved it to the bottom as a separate test.

File: client/src/components/Section.test.js (hunk 2 above)
Comment (itssumertime) [reply client/src/components/Section.test.js]: Done.

File: client/src/components/Section.test.js (hunk 8 above)
Comment (itssumertime) [reply client/src/components/Section.test.js:99]: Isn't this already tested in Boolean and NumberQuestion? The response object is empty in all of these tests anyway and error text doesn't show up.

Comment (kebrahim) [reply client/src/components/Section.test.js:99]: yes, it's already tested in the questions, but we should also verify that the showError flag is correctly passed from Section to the questions

File: client/src/components/Section.test.js [hunk 9]
```diff
@@ -69,10 +63,37 @@ describe("section", () => {
...
     expect(radioNo).toBeInTheDocument();
   });
+
+  it("supports tab navigation", () => {
+    // Tab navigation cycles from body -> through questions
+    expect(document.body).toHaveFocus();
+    userEvent.tab(); // onto first question
+    const question1 = screen.getByRole("spinbutton", {
+      name: "HOUSEHOLD_SIZE.text",
+      id: "integer-question-HOUSEHOLD_SIZE",
```
Comment (kebrahim) [inline client/src/components/Section.test.js:79]: do you need to specify name, id & type? i think you might be able to only specify one of them (maybe just id or name)?

File: client/src/components/Section.test.js [hunk 10]
```diff
@@ -69,10 +63,37 @@ describe("section", () => {
...
+      type: "number",
+    });
+    expect(question1).toHaveFocus();
+    userEvent.tab(); // onto second question
+    const question2 = screen.getByLabelText("radio.yes");
+    expect(question2).toHaveFocus();
+    userEvent.tab(); // onto third question
+    const question3 = screen.getByRole("spinbutton", {
+      name: "HOUSEHOLD_INCOME.text",
+      id: "integer-question-INCOME",
```
Comment (kebrahim) [inline client/src/components/Section.test.js:89]: same comment about only needing to specify one key

File: client/src/components/Section.test.js [hunk 11]
```diff
@@ -69,10 +63,37 @@ describe("section", () => {
...
+      type: "number",
+    });
+    expect(question3).toHaveFocus();
```
Comment (kebrahim) [inline client/src/components/Section.test.js:92]: add one more `userEvent.tab()` & confirm that document.body has focus

File: client/src/components/Section.test.js [hunk 12]
```diff
@@ -69,10 +63,37 @@ describe("section", () => {
...
+  });
+});
+
+test("has no accessibility violations", async () => {
+  const { container } = renderWith(<Section />);
```
Comment (kebrahim) [inline client/src/components/Section.test.js:97]: mind also passing in the section & question keys? this test is just going to render an empty section

File: client/src/components/Section.test.js (hunk 9 above)
Comment (marcja) [reply client/src/components/Section.test.js:79]: just use name

File: client/src/components/Section.test.js (hunk 11 above)
Comment (marcja) [reply client/src/components/Section.test.js:92]: this seems brittle -- why add this?

Comment (itssumertime) [reply client/src/components/Section.test.js:88]: Added for accessibility testing via tabs.

File: client/src/components/Section.test.js (hunk 12 above)
Comment (itssumertime) [reply client/src/components/Section.test.js:117]: Yes done.

File: client/src/components/Section.test.js [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+        ADULTS_65_PLUS: {
+          type: "BOOLEAN",
+        },
+        HOUSEHOLD_INCOME: {
+          type: "NUMBER",
+        },
+      },
+    };
+    const sectionKey = "HOUSEHOLD_SECTION";
+    const questionKeys = [
```
[the remark this answers is not in the document]
Comment (itssumertime) [reply client/src/components/Section.test.js:36]: It's not multiple sections (I thought so at first too since I used data from other tests). It's all the household section (adults 65+ is part of ages of household).

File: client/src/components/Section.test.js [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+
+  it("renders heading", async () => {
+    const heading = screen.getByRole("heading", {
+      name: "HOUSEHOLD_SECTION.title",
+    });
+    expect(heading).toBeInTheDocument();
+  });
+
+  it("renders questions", async () => {
+    const questionOne = screen.getByText("ADULTS_65_PLUS.text");
```
[the remark this answers is not in the document]
Comment (itssumertime) [reply client/src/components/Section.test.js:62]: Those don't work for the ones in div classes but I did use the roles for spinButtons.

The error for switching this to getByLabelText is: "Found a label with the text of: ADULTS_65_PLUS.text, however no form control was found associated to that label. Make sure you're using the "for" attribute or "aria-labelledby" attribute correctly." I think our formatting may be slightly different.

File: client/src/components/Section.test.js (hunk 8 above)
Comment (itssumertime) [reply client/src/components/Section.test.js:128]: Done.
